ClientServerTest: Compile issue in `clientserver_getservices.cpp`

### Problem Description

Something seems to have changed in the API which prevents the client_server test from compiling. The failing cpp file didn't exist in 5.11, so I assume there hasn't been any API break, but we just forgot to adapt that file to some changes we made in between releases.

```
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(31): error C2039: 'SServiceMethodInfo': is not a member of 'eCAL::Util'
2>D:\Projects\ecal\ecal-githubcom-2\ecal\core\include\ecal/ecal_util.h(40): note: see declaration of 'eCAL::Util'
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(31): error C2065: 'SServiceMethodInfo': undeclared identifier
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(31): error C2923: 'std::map': 'SServiceMethodInfo' is not a valid template type argument for parameter '_Ty'
2>C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\map(70): error C3203: 'pair': unspecialized class template can't be used as a template argument for template parameter '_Ty', expected a real type (compiling source file D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp)
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(31): error C2133: 'service_info_map': unknown size
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(31): error C2512: 'std::map': no appropriate default constructor available
2>C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\map(72): note: see declaration of 'std::map' (compiling source file D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp)
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(42): error C2664: 'void eCAL::Util::GetServices(std::map<std::tuple<std::string,std::string>,eCAL::SServiceMethodInformation,std::less<_Kty>,std::allocator<std::pair<const _Kty,_Ty>>> &)': cannot convert argument 1 from 'std::map' to 'std::map<std::tuple<std::string,std::string>,eCAL::SServiceMethodInformation,std::less<_Kty>,std::allocator<std::pair<const _Kty,_Ty>>> &'
2>        with
2>        [
2>            _Kty=std::tuple<std::string,std::string>,
2>            _Ty=eCAL::SServiceMethodInformation
2>        ]
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(45): error C2662: '_Tree_comp_alloc<_Traits>::_Val_types::size_type std::_Tree<_Traits>::size(void) noexcept const': cannot convert 'this' pointer from 'std::map' to 'const std::_Tree<_Traits> &'
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(45): note: Reason: cannot convert from 'std::map' to 'const std::_Tree<_Traits>'
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(45): note: Conversion requires a second user-defined-conversion operator or constructor
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(45): error C2660: 'testing::internal::EqHelper::Compare': function does not take 3 arguments
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(45): error C2512: 'testing::AssertionResult': no appropriate default constructor available
2>D:\Projects\ecal\ecal-githubcom-2\thirdparty\googletest\googletest\include\gtest/gtest.h(283): note: see declaration of 'testing::AssertionResult' (compiling source file D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp)
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(51): error C2664: 'void eCAL::Util::GetServices(std::map<std::tuple<std::string,std::string>,eCAL::SServiceMethodInformation,std::less<_Kty>,std::allocator<std::pair<const _Kty,_Ty>>> &)': cannot convert argument 1 from 'std::map' to 'std::map<std::tuple<std::string,std::string>,eCAL::SServiceMethodInformation,std::less<_Kty>,std::allocator<std::pair<const _Kty,_Ty>>> &'
2>        with
2>        [
2>            _Kty=std::tuple<std::string,std::string>,
2>            _Ty=eCAL::SServiceMethodInformation
2>        ]
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(54): error C2662: '_Tree_comp_alloc<_Traits>::_Val_types::size_type std::_Tree<_Traits>::size(void) noexcept const': cannot convert 'this' pointer from 'std::map' to 'const std::_Tree<_Traits> &'
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(54): note: Reason: cannot convert from 'std::map' to 'const std::_Tree<_Traits>'
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(54): note: Conversion requires a second user-defined-conversion operator or constructor
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(54): error C2660: 'testing::internal::EqHelper::Compare': function does not take 3 arguments
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(54): error C2512: 'testing::AssertionResult': no appropriate default constructor available
2>D:\Projects\ecal\ecal-githubcom-2\thirdparty\googletest\googletest\include\gtest/gtest.h(283): note: see declaration of 'testing::AssertionResult' (compiling source file D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp)
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(61): error C2664: 'void eCAL::Util::GetServices(std::map<std::tuple<std::string,std::string>,eCAL::SServiceMethodInformation,std::less<_Kty>,std::allocator<std::pair<const _Kty,_Ty>>> &)': cannot convert argument 1 from 'std::map' to 'std::map<std::tuple<std::string,std::string>,eCAL::SServiceMethodInformation,std::less<_Kty>,std::allocator<std::pair<const _Kty,_Ty>>> &'
2>        with
2>        [
2>            _Kty=std::tuple<std::string,std::string>,
2>            _Ty=eCAL::SServiceMethodInformation
2>        ]
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(64): error C2662: '_Tree_comp_alloc<_Traits>::_Val_types::size_type std::_Tree<_Traits>::size(void) noexcept const': cannot convert 'this' pointer from 'std::map' to 'const std::_Tree<_Traits> &'
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(64): note: Reason: cannot convert from 'std::map' to 'const std::_Tree<_Traits>'
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(64): note: Conversion requires a second user-defined-conversion operator or constructor
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(64): error C2660: 'testing::internal::EqHelper::Compare': function does not take 3 arguments
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(64): error C2512: 'testing::AssertionResult': no appropriate default constructor available
2>D:\Projects\ecal\ecal-githubcom-2\thirdparty\googletest\googletest\include\gtest/gtest.h(283): note: see declaration of 'testing::AssertionResult' (compiling source file D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp)
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(80): error C2664: 'void eCAL::Util::GetServices(std::map<std::tuple<std::string,std::string>,eCAL::SServiceMethodInformation,std::less<_Kty>,std::allocator<std::pair<const _Kty,_Ty>>> &)': cannot convert argument 1 from 'std::map' to 'std::map<std::tuple<std::string,std::string>,eCAL::SServiceMethodInformation,std::less<_Kty>,std::allocator<std::pair<const _Kty,_Ty>>> &'
2>        with
2>        [
2>            _Kty=std::tuple<std::string,std::string>,
2>            _Ty=eCAL::SServiceMethodInformation
2>        ]
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(83): error C2662: '_Tree_comp_alloc<_Traits>::_Val_types::size_type std::_Tree<_Traits>::size(void) noexcept const': cannot convert 'this' pointer from 'std::map' to 'const std::_Tree<_Traits> &'
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(83): note: Reason: cannot convert from 'std::map' to 'const std::_Tree<_Traits>'
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(83): note: Conversion requires a second user-defined-conversion operator or constructor
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(83): error C2660: 'testing::internal::EqHelper::Compare': function does not take 3 arguments
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(83): error C2512: 'testing::AssertionResult': no appropriate default constructor available
2>D:\Projects\ecal\ecal-githubcom-2\thirdparty\googletest\googletest\include\gtest/gtest.h(283): note: see declaration of 'testing::AssertionResult' (compiling source file D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp)
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(124): error C2664: 'void eCAL::Util::GetServices(std::map<std::tuple<std::string,std::string>,eCAL::SServiceMethodInformation,std::less<_Kty>,std::allocator<std::pair<const _Kty,_Ty>>> &)': cannot convert argument 1 from 'std::map' to 'std::map<std::tuple<std::string,std::string>,eCAL::SServiceMethodInformation,std::less<_Kty>,std::allocator<std::pair<const _Kty,_Ty>>> &'
2>        with
2>        [
2>            _Kty=std::tuple<std::string,std::string>,
2>            _Ty=eCAL::SServiceMethodInformation
2>        ]
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(127): error C2662: '_Tree_comp_alloc<_Traits>::_Val_types::size_type std::_Tree<_Traits>::size(void) noexcept const': cannot convert 'this' pointer from 'std::map' to 'const std::_Tree<_Traits> &'
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(127): note: Reason: cannot convert from 'std::map' to 'const std::_Tree<_Traits>'
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(127): note: Conversion requires a second user-defined-conversion operator or constructor
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(127): error C2660: 'testing::internal::EqHelper::Compare': function does not take 3 arguments
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(127): error C2512: 'testing::AssertionResult': no appropriate default constructor available
2>D:\Projects\ecal\ecal-githubcom-2\thirdparty\googletest\googletest\include\gtest/gtest.h(283): note: see declaration of 'testing::AssertionResult' (compiling source file D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp)
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(134): error C2664: 'void eCAL::Util::GetServices(std::map<std::tuple<std::string,std::string>,eCAL::SServiceMethodInformation,std::less<_Kty>,std::allocator<std::pair<const _Kty,_Ty>>> &)': cannot convert argument 1 from 'std::map' to 'std::map<std::tuple<std::string,std::string>,eCAL::SServiceMethodInformation,std::less<_Kty>,std::allocator<std::pair<const _Kty,_Ty>>> &'
2>        with
2>        [
2>            _Kty=std::tuple<std::string,std::string>,
2>            _Ty=eCAL::SServiceMethodInformation
2>        ]
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(137): error C2662: '_Tree_comp_alloc<_Traits>::_Val_types::size_type std::_Tree<_Traits>::size(void) noexcept const': cannot convert 'this' pointer from 'std::map' to 'const std::_Tree<_Traits> &'
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(137): note: Reason: cannot convert from 'std::map' to 'const std::_Tree<_Traits>'
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(137): note: Conversion requires a second user-defined-conversion operator or constructor
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(137): error C2660: 'testing::internal::EqHelper::Compare': function does not take 3 arguments
2>D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp(137): error C2512: 'testing::AssertionResult': no appropriate default constructor available
2>D:\Projects\ecal\ecal-githubcom-2\thirdparty\googletest\googletest\include\gtest/gtest.h(283): note: see declaration of 'testing::AssertionResult' (compiling source file D:\Projects\ecal\ecal-githubcom-2\testing\ecal\clientserver_test\src\clientserver_getservices.cpp)
2>Done building project "test_clientserver.vcxproj" -- FAILED.
```

### How to reproduce

1. Check out eCAL master or support/v5.12
2. Activate the clientserver test in the main `CMakeLists.txt`
3. compile

### How did you get eCAL?

Custom Build / Built from source

### Environment

- eCAL Master
- Windows
- VS 2019

### eCAL System Information

_No response_

--- a/testing/ecal/clientserver_test/src/clientserver_getservices.cpp
+++ b/testing/ecal/clientserver_test/src/clientserver_getservices.cpp
@@ -18,6 +18,7 @@
 */
 
 #include <ecal/ecal.h>
+#include <ecal/ecal_types.h>
 
 #include <gtest/gtest.h>
 
@@ -28,7 +29,7 @@ TEST(IO, GetServices)
   // initialize eCAL API
   eCAL::Initialize(0, nullptr, "clientserver_getservices");
 
-  std::map<std::tuple<std::string, std::string>, eCAL::Util::SServiceMethodInfo> service_info_map;
+  std::map<std::tuple<std::string, std::string>, eCAL::SServiceMethodInformation> service_info_map;
   
   // add and expire simple service
   {

